OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/Optimobile.java

```diff
@@ -118,6 +118,23 @@ public static synchronized void initialize(final Application application, Optimo
         maybeMigrateUserAssociation(application, customerId);
     }
 
+    public static synchronized void completeDelayedConfiguration(Context context, OptimoveConfig config){
+        if (!config.usesDelayedOptimobileConfiguration()){
+            throw new IllegalStateException("Trying to complete optimobile init without using delayed configuration");
+        }
+        authHeader = buildBasicAuthHeader(config.getApiKey(), config.getSecretKey());
+
+        flushEvents(context);
+        maybeTriggerInAppSync(context);
+        if (config.getDeferredDeepLinkHandler() != null){
+            deepLinkHelper.maybeProcessCachedLink(context);
+        }
+    }
+
+    static boolean hasFinishedHttpInitialisation(){
+        return authHeader != null;
+    }
```

Comment:
It feels like this object has too many responsibilities. Can we reduce them somehow? At least this one?

Reply:
yeah, i moved authHeader to the network client